server/server.go

```diff
@@ -217,56 +218,48 @@ func (s *Server) Listeners() (func() error, func() error) {
 	return loop2, loop1
 }
 
-func (s *Server) execQuery(ctx context.Context, compiler *ast.Compiler, txn storage.Transaction, query ast.Body, input ast.Value, explainMode types.ExplainModeV1) (types.QueryResponseV1, error) {
-
-	t := topdown.New(ctx, query, s.Compiler(), s.store, txn).WithInput(input)
-
+func (s *Server) execQuery(ctx context.Context, query string, input ast.Value, explainMode types.ExplainModeV1, includeMetrics bool) (results types.QueryResponseV1, err error) {
 	var buf *topdown.BufferTracer
-
 	if explainMode != types.ExplainOffV1 {
 		buf = topdown.NewBufferTracer()
-		t.Tracer = buf
 	}
+	m := metrics.New()
 
-	qrs := types.AdhocQueryResultSetV1{}
-
-	err := topdown.Eval(t, func(t *topdown.Topdown) error {
-		result := map[string]interface{}{}
-		for k, v := range t.Vars() {
-			if !k.IsWildcard() {
-				x, err := ast.ValueToInterface(v, t)
-				if err != nil {
-					return err
-				}
-				result[k.String()] = x
-			}
-		}
-		if len(result) > 0 {
-			qrs = append(qrs, result)
-		}
-		return nil
-	})
-
+	compiler := s.Compiler()
+	r := rego.New(
+		rego.Store(s.store),
+		rego.Compiler(compiler),
+		rego.Query(query),
+		rego.Input(input),
+		rego.Metrics(m),
+		rego.Tracer(buf),
+	)
+
+	output, err := r.Eval(ctx)
 	if err != nil {
-		return types.QueryResponseV1{}, err
```

Comment:
Whoops, I commented too quickly on this. I saw that the code was refactoring to use the rego package and I just assumed the return type was being changed. My bad.